srml/contract/src/account_db.rs

```diff
@@ -123,7 +123,7 @@ impl<T: Trait> AccountDb<T> for DirectAccountDb {
 						storage_size: <Module<T>>::storage_size_offset(),
 						trie_id: <T as Trait>::TrieIdGenerator::trie_id(&address),
 						deduct_block: <system::Module<T>>::block_number(),
-						rent_allowance: <BalanceOf<T>>::zero(),
+						rent_allowance: <BalanceOf<T>>::max_value(),
```

Comment:
yes this should be override by the default value put in the overlay on create_new_code.

also there is https://github.com/paritytech/substrate/blob/82717cab42c5f87cefb37dc0ce19ecd9bdb60d1f/srml/contract/src/exec.rs#L644
it should never be triggered as contract always as a rent_allowance (it is in the same structure in the  storage and it is instantiated in the overlay on new contract)

Reply:
hmm maybe we should change also in exec.rs to make the default value equal Max ?